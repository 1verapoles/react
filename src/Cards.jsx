import React from 'react'
import Card from './Card'
import { useSelector } from 'react-redux'

function Cards() {
  const cards = useSelector(state => state.cards.articles)


  return (
    <div className="d-flex flex-wrap">
      {cards && cards.map((el, index) => <Card key={index} {...el} />)}
    </div>
  )
}


export default Cards
