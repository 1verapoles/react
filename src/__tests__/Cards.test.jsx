import React from 'react';
import { BrowserRouter } from "react-router-dom";
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../store';
import Cards from '../Cards';



describe('Testing Cards component', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(<Provider store={store}><BrowserRouter><Cards /></BrowserRouter></Provider>);
    expect(asFragment()).toMatchSnapshot();
  });
});