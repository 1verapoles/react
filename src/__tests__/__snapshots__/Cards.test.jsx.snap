// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Cards component renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-wrap"
  />
</DocumentFragment>
`;
